controllers/workflow_controller.go

```diff
@@ -27,15 +31,49 @@ func (r *DSPAReconciler) ReconcileWorkflowController(dsp *dspav1.DataSciencePipe
 
 	log := r.Log.WithValues("namespace", dsp.Namespace).WithValues("dspa_name", dsp.Name)
 
-	if dsp.Spec.WorkflowController == nil || !dsp.Spec.WorkflowController.Deploy {
-		log.Info("Skipping Application of WorkflowController Resources")
-		return nil
-	}
+	// Get the management state for the WorkflowController subcomponent from the config
+	// Expected format example: {"managementState":"Managed"}
+	dspoArgoWorkflowsControllersJSON := config.GetStringConfigWithDefault("DSPO.ArgoWorkflowsControllers", config.DefaultArgoWorkflowsControllers)
 
-	log.Info("Applying WorkflowController Resources")
+	argoWorkflowsControllersConfig := map[string]string{}
+	var argoWorkflowsControllerManagementState string
 
-	err := r.ApplyDir(dsp, params, workflowControllerTemplatesDir)
+	err := json.Unmarshal([]byte(dspoArgoWorkflowsControllersJSON), &argoWorkflowsControllersConfig)
 	if err != nil {
+		log.Info(fmt.Sprintf("Unable to parse Argo Workflows Controller management state, using default value: %s", config.DefaultArgoWorkflowsControllersManagementState))
+		argoWorkflowsControllerManagementState = config.DefaultArgoWorkflowsControllersManagementState
+	} else {
+		argoWorkflowsControllerManagementState = argoWorkflowsControllersConfig["managementState"]
+	}
+
+	// Conditionally deploy the WorkflowController resource depending on the speciified management state
+	// Managed (or blank) - deploy the WorkflowController subcomponent
+	// Removed - skip deploying, and remove if already present, the WorkflowController subcomponent
+	// All other values - Invalid configuration, return an error
+	switch argoWorkflowsControllerManagementState {
+	case "Managed", "":
+
+		if dsp.Spec.WorkflowController == nil || !dsp.Spec.WorkflowController.Deploy {
```

Comment:
I think the behavior of `dsp.Spec.WorkflowController.Deploy` is confusing in that it overrides `Managed` but not `Removed`.

I think we should be consistent in terms of priority and document the behavior on the CRD description.

Reply:
yes - because `Deploy` is a bool that defaults to `true` we have to ignore it for the Removed case, or else the global configuration switch would do nothing (even it was set to `Removed`, all the DSPAs would still have Deploy: true and therefore still would deploy WCs).  I think updating the CRD description is the best bet, especially since this is considered a 'development/testing' field only